app/src/main/java/org/schabi/newpipe/util/ShareUtils.java

```diff
@@ -2,19 +2,77 @@
 
 import android.content.Context;
 import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.content.pm.ResolveInfo;
 import android.net.Uri;
 
 import org.schabi.newpipe.R;
 
 public final class ShareUtils {
-    private ShareUtils() { }
+    private ShareUtils() {
+    }
 
+    /**
+     * Open the url with the system default browser.
+     * <p>
+     * If no browser is set as default, fallbacks to
+     * {@link ShareUtils#openUrlBrowserChooser(Context, String)}
+     *
+     * @param context the context to use
+     * @param url     the url to browse
+     */
     public static void openUrlInBrowser(final Context context, final String url) {
-        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+        final Intent intent = new Intent(Intent.ACTION_VIEW);
+        intent.setData(Uri.parse(url));
+
+        final String defaultBrowserPackageName = getDefaultBrowserPackageName(context);
+
+        if (defaultBrowserPackageName.equals("android")) {
+            // no browser set as default
+            openUrlBrowserChooser(context, url);
+        } else {
+            intent.setPackage(defaultBrowserPackageName);
+            context.startActivity(intent);
+        }
+    }
+
+    /**
+     * Open the url with application chooser, including browser and apps able to open url.
+     * <p>
+     * If any app (except browser, typically NewPipe) is set as default,
+     * it will nor open in browser, neither open the chooser, but just the default app.
+     *
+     * @param context the context to use
+     * @param url     the url to browse
+     */
+    private static void openUrlBrowserChooser(final Context context, final String url) {
+        final Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
         context.startActivity(Intent.createChooser(
                 intent, context.getString(R.string.share_dialog_title)));
```

Comment:
Maybe the title should be changed to something like "Open in browser"

Reply:
Nope, that's why i did this PR: it doesn't open in browser, but in favorite / default app for the type of link (in NewPipe case, "open in Browser" doesn't open browser but newpipe, so it's pointless) , and if there is no default app set for this type of link, it opens a chooser (with browser and NewPipe).

What about `openInDefaultApp()`?

Reply:
Oh right sorry ;-)
Yeah, `openInDefaultApp()` is maybe a better name